source/fs/fsutil.c

```diff
@@ -525,11 +525,9 @@ bool PathCopyVrtToFat(char* dest, char* orig, u32* flags) {
         }
     }
     
-    // check destination write permission (special paths only)
-    if (((*dest == '1') || (strncmp(dest, "0:/Nintendo 3DS", 16) == 0)) &&
-        (!flags || (*flags & (OVERWRITE_CUR|OVERWRITE_ALL))) &&
-        (!flags || !(*flags & OVERRIDE_PERM)) && 
-        !CheckWritePermissions(dest)) return false;
+    // check destination write permission
+    if ((!flags || !(*flags & OVERRIDE_PERM)) && !CheckWritePermissions(dest))
+        return false;
     
```

Comment:
Yup, I noticed. To be completely honest, I don't even know if checking for each and every file would cost that much time. Better not to do it, though, if we can prevent it. In any case, thank you for these improvements! If you want your name on that second commit, I'd need your Github email so that I can attribute it back to you. You can also send this in a pm, though.